hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/CustomTypeFactory.java

```diff
@@ -236,4 +249,29 @@ public static Schema createSchema(String typeName, List<FieldDescriptor> fields)
     public static HazelcastJsonValue createHazelcastJsonValue(String value) {
         return new HazelcastJsonValue(value);
     }
+
+    public static MemoryTierConfig createMemoryTierConfig(long capacityValue, int capacityMemoryUnitType) {
+        MemoryTierConfig config = new MemoryTierConfig();
+        config.setCapacity(Capacity.of(capacityValue, MemoryUnit.values()[capacityMemoryUnitType]));
```

Comment:
Regarding `MemoryUnit.values()[capacityMemoryUnitType])`, for enums we use in the protocol, we generally make sure that there are fixed ids(or names if you use strings to represent them) and we have a test case for them in `EnumCompatibilityTest`.

So, can we add a field to `MemoryUnit` to represent ids for them and add necessary methods to it so that we can construct the enum member back from the id?